client/components/tinymce/plugins/contact-form/dialog.jsx

```diff
@@ -0,0 +1,103 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import Shortcode from 'lib/shortcode';
+import Dialog from 'components/dialog';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormButton from 'components/forms/form-button';
+
+/**
+ * object constants
+ */
+const fieldTypes = {
+	name: 'name',
+	email: 'email',
+	url: 'url',
+	checkbox: 'checkbox',
+	dropdown: 'dropdown',
+	radio: 'radio',
+	text: 'text',
+	textarea: 'textarea',
+	website: 'website'
+};
+
+const defaultForm = [
+	{ label: 'Name', type: fieldTypes.name, required: true },
+	{ label: 'Email', type: fieldTypes.email, required: true },
+	{ label: 'Website', type: fieldTypes.url },
+	{ label: 'Comment', type: fieldTypes.textarea, required: true }
+];
+
+export default React.createClass( {
+	displayName: 'ContactFormDialog',
+
+	getInitialState() {
+		return {
+			isVisible: true
+		};
+	},
+
+	componentWillReceiveProps( nextProps ) {
+		if ( nextProps.visible && ! this.state.isVisible ) {
+			this.setState( { isVisible: true } );
+		}
+	},
+
+	render() {
+		const buttons = [
+			<FormButton
+				key="save"
+				onClick={ () => {
+					let fields = [];
+					for ( let field of defaultForm ) {
```

Comment:
In general, I try to avoid `for` loops as much as possible, as there's usually a more expressive means of writing the same intent. Here, I'd recommend using [`map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map).

```jsx
const fields = defaultForm.map( ( field ) => {
	return Shortcode.stringify( {
		tag: 'contact-field',
		type: 'self-closing',
		attrs: {
			label: field.label,
			type: field.type,
			required: field.required ? 1 : 0
		}
	} );
} );
```